app/helpers/sessions_helper.rb

```diff
@@ -3,15 +3,36 @@ def log_in user
     session[:user_id] = user.id
   end
 
+  def remember user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
   def current_user
-    @current_user ||= User.find_by(id: session[:user_id]) if session[:user_id]
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by id: user_id
+    elsif (user_id = cookies.signed[:user_id])
+      user = User.find_by id: user_id
+      if user&.authenticated?(cookies[:remember_token])
```

Comment:
user&.authenticated?(cookies[:remember_token]) 
Dấu & để làm gì vậy bn???

Reply:
Bn đọc thêm để hiểu kỹ hơn nhé http://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/